pageserver/src/deletion_queue/list_writer.rs

```diff
@@ -195,7 +197,7 @@ impl ListWriter {
                     debug!("Deletion header {header_path} not found, first start?");
                     Ok(None)
                 } else {
-                    Err(anyhow::anyhow!(e))
+                    on_fatal_io_error(&e);
```

Comment:
Wouldn't it be better to do `Err(Error::new(e))?` here?

And if not, the function doesn't have to return a Result.